docs/source/user_guides/basic_conceptions.rst

```diff
@@ -139,165 +148,185 @@ That's what the agent consists of:
   no transition for the message given by user is found.
   It transfers user to ``greeting_node`` no matter what user writes.
 
-* ``pipeline`` is a special object that processes user requests according to provided script.
-  In order to create a pipeline, the script should be provided and two two-string tuples:
-  the first specifies initial node flow and name and the second (optional) specifies fallback
-  node flow and name (if not provided it equals to the first one by default). 
+* ``pipeline`` is a special object that traverses the script graph based on the values of user input.
+  It is also capable of executing custom actions that you want to run on every turn of the conversation.
+  The pipeline can be initialized with a script, and with labels of two nodes:
+  the entrypoint of the graph, aka the 'start node', and the fallback node
+  (if not provided it equals the former per default).
 
 .. note::
 
     See `tutorial on basic dialog structure`_.
 
-Advanced graph features
-~~~~~~~~~~~~~~~~~~~~~~~
+Processing Definition
+~~~~~~~~~~~~~~~~~~~~~
+
+Processing user requests and extracting additional parameters is a crucial part of building a conversational bot. 
+DFF allows you to define how user requests will be processed to extract additional parameters.
+This is done by passing callbacks to a special ``PROCESSING`` fields in a Node dict.
+
+* User input can be altered with ``PRE_RESPONSE_PROCESSING`` and will happen **before** response generation. See `tutorial on pre-response processing`_.
+* Node response can be modified with ``PRE_TRANSITIONS_PROCESSING`` and will happen **after** response generation. See `tutorial on pre-transition processing`_.
 
-Right now the agent we have created is a very simple one and does not behave **exactly** as we wanted
-our bot to behave. Let's see how we can improve our script:
+Depending on your bot's requirements and the goal of the dialog, you may need to interact with external databases or APIs to retrieve data. 
+For instance, if a user wants to know a schedule, you may need to access a database and extract parameters such as date and location.
 
 .. code-block:: python
 
-    from dff.pipeline import Pipeline
-    from dff.script import TRANSITIONS, RESPONSE, Context, Message
-    import dff.script.conditions as cnd
-    import dff.script.labels as lbl
+    import requests
+    ...
+    def use_api_processing(ctx: Context, _: Pipeline) -> Context:
+        ctx.misc["api_call_results"] = requests.get("http://schedule.api/day1")
```

Comment:
I don't think that `Request` object are serializable and, as such, trying to store this context in a db would break the bot.
I think we should include some warning about it.

Reply:
Updated